### DISCUSSION TITLE
btstack license

### DISCUSSION THREAD

**Discussion by edy14431** [micropython/micropython#12040] [Libraries & Drivers] [answered]
I would like to use micropython firmware on a raspberry pi pico that will be used in a commercial project. Micropython is under the MIT license and therefore can be used commercially, however the BTstack library, which is mentioned in the Micropython license, is under its own license that does not allow commercial use (https://github.com/bluekitchen/ btstack/blob/77e752abd6a0992334047a48038a5a3960e5c6bc/LICENSE). How to understand this conflict? Is BTstack installed along with the firmware? Is it an external firmware dependency? When using the firmware in a commercial project do I need to consider both licenses?

**Comment by bidrohini** [comment]
When using Micropython firmware on a Raspberry Pi Pico, you need to consider the licenses of all the components that come with it, including external libraries like BTstack.

The BTstack library is a separate project, and it appears that it is not directly bundled with Micropython but is mentioned in the Micropython license as one of the dependencies used with Micropython. BTstack has its own license, which you have identified as not allowing commercial use.

To understand this conflict, it's essential to clarify how BTstack is used in the context of Micropython. There are a few possibilities:

1. BTstack is an optional external dependency: If BTstack is not directly included in the Micropython firmware, and you have to separately install it or link to it during your project setup, then you need to adhere to the BTstack license separately. This means you cannot use it in a commercial project if its license disallows commercial use.

2. BTstack is part of the Micropython firmware distribution: If BTstack is bundled with the Micropython firmware distribution and distributed together, then you might have an issue with the commercial usage. In such a case, the conflict between the permissive MIT license of Micropython and the restrictive license of BTstack could cause licensing issues for commercial projects.

It's essential to carefully review the Micropython firmware distribution and its associated files to determine whether BTstack is included or if it is an external dependency that you need to handle separately. If BTstack is indeed included or required for the operation of Micropython on the Raspberry Pi Pico, and its license is not compatible with your commercial project, you may need to look for alternative libraries or consider seeking permission from the BTstack project maintainers for a different license agreement that allows commercial use.

Please note that licensing can be complex, and it's always a good idea to consult with legal counsel or licensing experts when dealing with such matters in a commercial context. This response is not legal advice, and you should seek professional advice for your specific situation.

**Reply by Josverl** [reply]
As your question is regarding btstack licensing, it's best to ask them directly

> BTstack is free for non-commercial use. However, for commercial use, tell us a bit about your project to get a quote.

**Reply by dlech** [reply]
Also, according to the RPi website, the Pico W has a pre-paid BTstack commercial license.

https://www.raspberrypi.com/news/new-functionality-bluetooth-for-pico-w/

**Reply by edy14431** [reply]
Thanks for information, guys! Actually I'm using raspberry pi pico without wireless or bluetooth, but if I'm not mistaken the firmware is the same in both cases. I'm worried about the BTstack licensing issue because I couldn't identify whether the standard firmware of the raspberry pi pico makes use of BTstack or not. If the firmware uses Btstack I think it would be necessary to pay the BTstack license.

**Reply by dlech** [reply]
You can always make your own build without it if you don't need it.

**Comment by scruss** [comment, marked as answer]
> … I couldn't identify whether the standard firmware of the raspberry pi pico makes use of BTstack or not.

It does not. It *can* not: there's no BT hardware on the regular Pico to call. 

To put your mind at ease, here's a string search for "BTstack" (case insensitve) in the Pico variant UF2 files:

## Pico W

```
strings rp2-pico-w-20230714-unstable-v1.20.0-283-g7d66ae603.uf2 | grep -i btstack
 MPY BTSTACK
 Please configure btstack_tlv to let BTstack setup ER and IR keys
```
So the Pico W code is likely to contain BTStack, but Raspberry Pi Ltd has already pre-paid for a commercial licence on this hardware.

## Pico

```
strings rp2-pico-20230714-unstable-v1.20.0-283-g7d66ae603.uf2 | grep -i btstack
```

There was no output from that command, thus it's very unlikely that BTStack is compiled in.

**Comment by jimmo** [comment]
> Actually I'm using raspberry pi pico without wireless or bluetooth, but if I'm not mistaken the firmware is the same in both cases.

@edy14431 No they are different firmware. Every board listed on micropython.org/download or built with a different `BOARD=` argument will result in different firmware.

@scruss Is correct -- the Pico firmware does not include the bluekitchen stack (nor does it include any bluetooth support at all).

The key detail is https://github.com/micropython/micropython/blob/master/ports/rp2/boards/PICO_W/mpconfigboard.cmake#L11 -- this is where the Pico W board definition enables btstack (as part of enabling bluetooth more broadly). This line is not present in other boards (e.g. the base Pico board).
